srml/staking/src/lib.rs

```diff
@@ -1399,22 +1410,42 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	/// Add reward points to validator.
+	/// Add reward points to validators. Using their stash account ID.
+	///
+	/// Validators are keyed by stash account ID and must be in the current elected set.
+	///
+	/// For each element in the iterator the given number of points in u32 is added to the
+	/// validator, thus duplicates are handled.
 	///
 	/// At the end of the era each the total payout will be distributed among validator
 	/// relatively to their points.
-	pub fn add_reward_points_to_validator(validator: T::AccountId, points: u32) {
-		<Module<T>>::current_elected().iter()
-			.position(|elected| *elected == validator)
-			.map(|index| {
-				CurrentEraRewards::mutate(|rewards| {
-					if let Some(new_total) = rewards.total.checked_add(points) {
-						rewards.total = new_total;
-						rewards.rewards.resize((index + 1).max(rewards.rewards.len()), 0);
-						rewards.rewards[index] += points; // Addition is less than total
+	pub fn add_reward_points(validators_points: Vec<(T::AccountId, u32)>) {
+		CurrentEraRewards::mutate(|rewards| {
+			for (index, elected) in <Module<T>>::current_elected().iter().enumerate() {
+				for (validator, points) in &validators_points {
```

Comment:
Might be worth looking into in order to remove the inner loop:
- If most modules are storing the indexes nowadays, if authorship can tell you reward validator at index `I` (instead of validator with address `A`) that would be nice.

Reply:
also I thought that if current elected is sorted we could do some binary_search.

in polkadot when rewarding validity statement we have indices

Reply:
oh I missunderstood, if you have index then use the other method add_reward_points_using_index